lib/honeycomb/propagation.rb

```diff
@@ -47,17 +47,35 @@ def parse_v1(payload)
 
       [trace_id, parent_span_id, trace_fields, dataset]
     end
+
+    private
+
+    def parse_span_id(string)
+      INVALID_SPAN_ID = ("0" * 16).b
+      raise "invalid span id" if string == INVALID_SPAN_ID
+
+      value.downcase!
+      Array(value).pack('H*')
+
+    def parse_trace_id(string)
+      INVALID_TRACE_ID = ("0" * 32).b
+      raise "invalid trace id" if string == INVALID_TRACE_ID
+
+      value.downcase!
+      Array(value).pack('H*')
   end
 
   # Serialize trace headers
   module PropagationSerializer
     def to_trace_header
       context = Base64.urlsafe_encode64(JSON.generate(trace.fields)).strip
       encoded_dataset = URI.encode_www_form_component(builder.dataset)
+      encoded_trace_id = trace.id.unpack('H*')
```

Comment:
Just trying to follow the existing pattern here but we could also just do this inline when defining `data_to_propagate`